polkadot/xcm/xcm-executor/src/lib.rs

```diff
@@ -1773,11 +1787,42 @@ impl<Config: config::Config> XcmExecutor<Config> {
 
 		// retry previously failed deposits, this time short-circuiting on any error.
 		for asset in failed_deposits {
-			Config::AssetTransactor::deposit_asset(&asset, &beneficiary, context)?;
+			let deposit_result =
+				Config::AssetTransactor::deposit_asset(&asset, &beneficiary, context);
+			// Gracefully skip dust deposits
+			if let Err(e) = &deposit_result {
+				if Self::is_below_minimum_error(&e) {
+					Self::do_handle_below_minimum(&asset, &beneficiary, context);
+
+					continue;
+				} else {
+					return Err(*e);
+				}
+			}
```

Comment:
on the second try, simply burn (ignore) any non-depositable dust

```suggestion
			if let Err(e) = Config::AssetTransactor::deposit_asset(&asset, &beneficiary, context) {
				// Ignore dust deposit errors.
				if !matches!(e, XcmError::FailedToTransactAsset(s) if *s == "BelowMinimum") {
					return Err(e);
				}
			}
```